x/profiles/keeper/msg_server_chain_link.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+func (k msgServer) LinkChainAccount(goCtx context.Context, msg *types.MsgLinkChainAccount) (*types.LinkChainAccountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	if err := msg.SourceProof.Verify(k.cdc); err != nil {
+		return nil, err
+	}
+
+	if err := msg.DestinationProof.Verify(k.cdc); err != nil {
+		return nil, err
+	}
+
+	// Check if address has the profile
+	profile, found, err := k.GetProfile(ctx, msg.DestinationAddress)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, fmt.Errorf("address does not have any profile")
+	}
```

Comment:
This should be moved inside the `StoreLink` method to make sure the checks are always performed even if the method is called from other places